app/deployment/remote_slave_service.py

```diff
@@ -20,6 +22,8 @@ def start(self, master_host, master_port, slave_port, num_executors):
         :param num_executors: the number of concurrent executors that will run in this slave service
         :type num_executors: int
         """
+        if master_host == 'localhost':
+            master_host = socket.gethostname()
```

Comment:
@josephharrington I thought about this as well. My original thought was to get rid of `localhost` completely, which is probably not going to happen since you want to have the capability to have `localhost` in the config yaml file to reference to the local host. And I find deployment the only place that this translation is only needed (I could totally be wrong). But anyway, if we find that there are some places else need to do this translation, we can think about how to do it better to keep it DRY. Does it sound good?